File: testers/avro/java-spring/generated-and-checked-in/com/example/events/DynamicValuePublisher.java
package com.example.events;

import java.util.concurrent.CompletableFuture;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.kafka.support.SendResult;
import org.springframework.stereotype.Service;

/** Type-safe event publisher for dynamic-value topic */
@Service
public record DynamicValuePublisher(
    KafkaTemplate<String, DynamicValue> kafkaTemplate, String topic) {
  public DynamicValuePublisher(KafkaTemplate<String, DynamicValue> kafkaTemplate) {
    this(kafkaTemplate, "dynamic-value");
  }

  public DynamicValuePublisher withKafkaTemplate(
      KafkaTemplate<String, DynamicValue> kafkaTemplate) {
    return new DynamicValuePublisher(kafkaTemplate, topic);
  }

  public DynamicValuePublisher withTopic(String topic) {
    return new DynamicValuePublisher(kafkaTemplate, topic);
  }

  /** Publish a DynamicValue event */
  public CompletableFuture<SendResult<String, DynamicValue>> publish(
      String key, DynamicValue event) {
    return kafkaTemplate.send(topic, key, event);
  }
}

File: testers/avro/scala-cats/generated-and-checked-in/com/example/events/DynamicValuePublisher.scala
package com.example.events

import cats.effect.IO
import fs2.kafka.KafkaProducer
import fs2.kafka.ProducerRecord
import fs2.kafka.ProducerRecords
import fs2.kafka.ProducerResult
import scala.annotation.unused

@unused
/** Type-safe event publisher for dynamic-value topic */
case class DynamicValuePublisher(
  kafkaTemplate: KafkaProducer[IO, String, DynamicValue],
  topic: String = "dynamic-value"
) {
  /** Publish a DynamicValue event */
  def publish(
    key: String,
    event: DynamicValue
  ): IO[ProducerResult[String, DynamicValue]] = {
    return kafkaTemplate.produce(ProducerRecords.one(ProducerRecord(topic, key, event))).flatten
  }
}